hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMKeyCommitRequest.java

```diff
@@ -263,10 +264,18 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager, TermIn
       omKeyInfo.getMetadata().putAll(KeyValueUtil.getFromProtobuf(
           commitKeyArgs.getMetadataList()));
 
+      // flag to indicate whether it is necessary to update the value (ozone key) in OpenKeyTable
+      boolean updateOpenKeyTable = false;
+
       if (isHSync) {
+        updateOpenKeyTable = !OmKeyHSyncUtil.isHSyncedPreviously(omKeyInfo, clientIdString, dbOpenKey);
         omKeyInfo.getMetadata().put(OzoneConsts.HSYNC_CLIENT_ID, clientIdString);
       } else if (isRecovery) {
+        // Q for Sammi/Wei-Chiu: The following line won't do anything (before this PR) because HSYNC_CLIENT_ID was never
+        // added to OpenKeyTable right? Have I missed something?
```

Comment:
Not really. The OMRecoverLeaseRequest will return either keyInfo or openKeyInfo depending on which one has the lastest block information, so there is chance that commitKeyArgs will have HSYNC_CLIENT_ID metadata.